lib/plugin.js

```diff
@@ -226,28 +232,58 @@ internals.implementation = function (server, options) {
 };
 
 
+const getTokenFromCookie = (cookieHeader, cookieName) => {
+
+    if (!(cookieHeader || cookieName)) {
+        return null;
+    }
+
+    const cookies = cookieHeader.split(/;(\s+)?/);
+
+    // find the cookie with the cookie name given in the settings
+    for (let i = 0; i < cookies.length; i += 1) {
+        const [name, value] = cookies[i].split('=');
+
+        if (name === cookieName) {
+            return value;
+        }
+    }
+
+    return '';
+};
+
+
 internals.token = function (request, settings, missing, unauthorized) {
 
-    const authorization = request.headers.authorization;
+    // if both the headerName and cookieName are given false then we should fallback to
+    // authorization header as the default
+    const authorization =
+        request.headers[settings.headerName]
+            || getTokenFromCookie(request.headers.cookie, settings.cookieName)
```

Comment:
I'm not a fan of the custom cookie header parsing. I would prefer that the existing hapi parser is used.

Reply:
How do you use the existing parser to parse the cookie header value ? Like we need to be able to read the field and its value from something like `token=ejTdsfsdhfsdjfsdjfsdfs.sdfsjdflkjsldfkjsdf.sdflsjdlfjlsdf; another=gibberish`, so that we get token field's value right ? Could give me some pointers on this ?

Reply:
https://hapi.dev/api?v=20.2.0#-serverstatename-options / https://hapi.dev/api?v=20.2.0#-await-serverstatesparseheader.

Reply:
You can also check out https://github.com/hapijs/cookie/blob/master/lib/index.js for an example of how another plugin handles cookies.

Reply:
ok I will read these and do the changes. This was something I had some doubts about earlier, so thank you for clearing that up

Reply:
Oh the cookies are already parsed for us to use 😃. I didn't know that